scripts/generator.py

```diff
@@ -58,7 +58,7 @@ def main():
     csv_generator.generate(flat, ecs_version, out_dir)
     es_template.generate(flat, ecs_version, out_dir)
     beats.generate(nested, ecs_version, out_dir)
-    asciidoc_fields.generate(nested, flat, ecs_version, docs_dir)
+    asciidoc_fields.generate(intermediate_fields, ecs_version, docs_dir)
```

Comment:
Why are you switching to passing `intermediate_fields` here?

The goal of the intermediate files (nested and flat) is to make the two structures possible to visualize, when passing it around to the other generators, and also for other external projects to have either file to start from, to generate other artifacts.

I'm all for no longer passing both "nested" and "flat" to asciidoc and passing just one instead, that was temporary a hack :-) But could we improve "nested" with what you're missing, instead of adding a third structure that we're passing around?

Reply:
Nested and flat are both missing the top level reused fields - i.e. `hash` is reused in `dll` so you can find `dll.hash.md5`, `dll.hash.sha1`, etc but `dll.hash` is not there.  To render the ascii doc reusable sections we look for the short description of the reusable fieldset, but in this example since `dll.hash` isn't in nested or flat we look for `hash` at the top level instead and get the short description there.  Unfortunately, this means if we are using a subset that includes `dll.hash` but doesn't include `hash` at the top level then ascii doc generation errors out.

I'm now realizing that in that example the link to the `hash` fieldset wouldn't work since those fields wouldn't be in the docs at the top level so the ascii docs are going to be broken in some cases with subsets anyway.  I'm not sure how high priority it is to get ascii docs fully working with subsets.

Reply:
The other reason I switched is that intermediate_fields is already being used as a data structure to generate the nested and flat structures.  My thought is that it will be simpler to use intermediate_fields as the input to all the generators since it has all the information and we can remove the dependency of the other generators on the nested and flat generator.

Reply:
In my mind, the asciidoc output when using --include or --subset is fraught with pitfalls and should be ignored.

E.g. in the case of --include, I think we should force set the field level to "custom" for everything that was included. Right now it takes whatever's in the included files as is. This has the potential to be very confusing for end users looking at custom docs.

The way I currently envision the output when using --include and/or --subset is that only the following files make sense to use without huge caveats or adjustments:

* the intermediate YAML files
* the Elasticsearch template
* the CSV

I should probably edit [this comment](https://github.com/elastic/ecs/issues/746#issuecomment-604013230) yet again, with these caveats.

Reply:
Sorry I got distracted for a moment.

IMO, fixing asciidoc when using --subset is not a priority :-)

Reply:
I think that makes sense, excluding the ascii docs when `--subset` is used makes it simpler.

I'd still like to use `intermediate_fields` as the input though so we can move towards using it as the only internal intermediate data structure.  Right now we essentially have 3 internal structures: `intermediate_fields`, `nested`, and `flat`, but `nested` and `flat` are generated from `intermediate_fields` and exposed in the generated docs as well.  In the overall code structure I think it's simpler to render the asciidocs directly from intermediate_fields rather than convert intermediate_fields to the nested/flat format and render the asciidocs from those - that additional flattening step adds complexity.  We can still keep `nested` and `flat` but present them as output docs that are stable to external users.